api/v1beta1/baremetalset_types.go

```diff
@@ -47,6 +52,70 @@ type BaremetalHostStatus struct {
 	Online                bool   `json:"online"`
 }
 
+// HardwareReqs defines request hardware attributes for the BaremetalHost replicas
+type HardwareReqs struct {
+	CPUReqs  CPUReqs  `json:"cpuReqs,omitempty"`
+	MemReqs  MemReqs  `json:"memReqs,omitempty"`
+	DiskReqs DiskReqs `json:"diskReqs,omitempty"`
+}
+
+// CPUReqs defines specific CPU hardware requests
+type CPUReqs struct {
+	// Arch is a scalar (string) because it wouldn't make sense to give it an "exact-match" option
+	Arch     string      `json:"arch,omitempty"`
```

Comment:
afaic right now only x86_64 and ppc64le computes are supported with OSP, should't we limit it to those archs?

https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/16.1/html-single/director_installation_and_usage/index#compute-node-requirements

Reply:
If that's indeed the case, that sounds reasonable to me